### PR TITLE
[Fundamental] Move devkit and azure tests

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Move devkit and azure tests.

Global Config Tests is pull request target tests.
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/operations/_local_azure_connection_operations.py [hunk 1]
```diff
@@ -37,7 +38,18 @@ def __init__(self, connection_provider, **kwargs):
     @property
     def _client(self):
         if self._pfazure_client is None:
-            from promptflow.azure._pf_client import PFClient as PFAzureClient
+            try:
+                from promptflow.azure._pf_client import PFClient as PFAzureClient
+            except ImportError as e:
+                msg = (
+                    '"promptflow[azure]" is required to validate trace provider, '
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/operations/_local_azure_connection_operations.py:45]: make add a constant as we used this multiple times

Comment (wangchao1230) [reply src/promptflow-devkit/promptflow/_sdk/operations/_local_azure_connection_operations.py:45]: @brynn-code I remember you have added something like this?

File: src/promptflow-tools/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
 google-search-results==2.4.1
-promptflow
```
Comment (wangchao1230) [inline src/promptflow-tools/requirements.txt:2]: MARK: wait promptflow tools team to delete this

Comment (crazygao) [reply src/promptflow-tools/requirements.txt:2]: Yes, pipelines also need to adapt.

Comment (crazygao) [reply src/promptflow-tools/requirements.txt:2]: let me remove this.

File: src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_cli.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 import mock
 import pytest
+from _constants import PROMPTFLOW_ROOT
```
Comment (wangchao1230) [inline src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_cli.py:20]: pls resolve conflict

**Review by brynn-code — DISMISSED** [review]

**Review by wangchao1230 — DISMISSED** [review]

**Review by brynn-code — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**